Public/Src/Engine/Scheduler/ChangeAffectedOutput/AffectedOutputList.cs

```diff
@@ -0,0 +1,258 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using BuildXL.Native.IO;
+using BuildXL.Pips;
+using BuildXL.Pips.Operations;
+using BuildXL.Scheduler.Artifacts;
+using BuildXL.Storage;
+using BuildXL.Storage.ChangeTracking;
+using BuildXL.Storage.InputChange;
+using BuildXL.Utilities;
+using BuildXL.Utilities.Collections;
+
+namespace BuildXL.Scheduler.ChangeAffectedOutput
+{
+    /// <summary>
+    /// Class representing source change affected output list.
+    /// </summary>
+    public sealed class AffectedOutputList
+    {
+
+        /// <summary>
+        /// Output file contents which affected by the source change of this build. 
+        /// </summary>
+        /// <remarks>
+        /// This list should be initialized by the input change list provided from the InputChanges configuration option.
+        /// </remarks>
+        private readonly ConcurrentBigSet<FileArtifact> m_sourceChangeAffectedOutputFiles = new ConcurrentBigSet<FileArtifact>();
+
+        /// <summary>
+        /// Output directories which affected by the source change of this build. 
+        /// </summary>
+        private readonly ConcurrentBigSet<DirectoryArtifact> m_sourceChangeAffectedOutputDirectories = new ConcurrentBigSet<DirectoryArtifact>();
+
+        private PathTable m_pathTable;
+
+        private FileContentManager m_fileContentManager;
+
+        /// <nodoc />
+        public AffectedOutputList(PathTable pathTable, FileContentManager fileContentManager)
+        {
+            m_pathTable = pathTable;
+            m_fileContentManager = fileContentManager;
+        }
+
+        /// <summary>
+        /// Initial affected output list with the external source change list
+        /// </summary>
+        public bool InitialAffectedOutputList(InputChangeList inputChangeList, PathTable pathTable)
+        {
+            var allInputsAreAffected = false;
+            // The Algorithm can't deal with removal currently, so disable change-based code coverage and have to run full code coverage if has removal. 

```

Comment:
>change-based code coverage and have to run full code coverage [](start = 75, length = 61)

nit - change based code coverage is the first consumer of this but it is meant to be a generic feature. so make the comments generic

Reply:
done